frontend/src/recoil_state.tsx

```diff
@@ -0,0 +1,17 @@
+import { atom } from "recoil";
+
+const initialStates = {
```

Comment:
You may want to explicitly type this as being `: ButtonStates` so that the atom also carries that interface